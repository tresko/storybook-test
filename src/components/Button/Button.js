import React from 'react';
import PropTypes from 'prop-types';

// Styles
import ButtonStyle from './Button.styles';

const propTypes = {
  /** onClick handler */
  onClick: PropTypes.func,

  /** button text. */
  text: PropTypes.string.isRequired,
};

const defaultProps = {
  onClick: () => {},
};

function Button({ text, onClick }) {
  return (
    <ButtonStyle onClick={onClick}>
      { text }
    </ButtonStyle>
  );
}

Button.propTypes = propTypes;
Button.defaultProps = defaultProps;

export default Button;