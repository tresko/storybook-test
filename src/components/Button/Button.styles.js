import styled from 'styled-components';

const ButtonStyle = styled.button`
  background: palevioletred;
  border: 0;
  width: auto;
  padding: 10px 20px;
  color: white;
  font-weight: bold;
  font-size: 14px;
`;

export default ButtonStyle;