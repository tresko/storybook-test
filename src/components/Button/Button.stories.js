import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { withKnobs, text } from '@storybook/addon-knobs';
import { withNotes } from '@storybook/addon-notes';
import { withInfo } from '@storybook/addon-info';
import { checkA11y } from '@storybook/addon-a11y';
import backgrounds from "@storybook/addon-backgrounds";

import Button from './Button';

storiesOf('Button Custom', module)
  .addDecorator(withKnobs)
  .addDecorator(backgrounds([
    { name: "light blue", value: "#a0fcff", default: true },
    { name: "twitter", value: "#00aced" },
    { name: "facebook", value: "#3b5998" },
  ]))
  .addDecorator(checkA11y)
  .add(
    'default',
    withNotes('A very simple button')(
      () => <Button onClick={action('clicked')} text={text('Button text', 'Hello Button')} />
    )
  )
  .add('simple info',
    withInfo(`
      description or documentation about my Button, supports markdown
    
      ~~~js
      <Button
        onClick={() => {}}
        text="Test"
      />
      ~~~
    
    `)(() =>
      <Button onClick={action('clicked')} text={text('Button text', 'Hello Button')} />
    )
  )
