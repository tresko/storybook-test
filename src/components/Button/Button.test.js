/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

// Components
import Button from './Button';

const test = describe('Button component', () => {
    it('should render Button with text', () => {
        const component = renderer.create(<Button text="test" />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render Button with text and onClick handler', () => {
        const onClickHandler = jest.fn();
        const wrapper = shallow(<Button text="test" onClick={onClickHandler} />);

        wrapper.simulate('click');
        expect(onClickHandler).toBeCalled();
    });
});