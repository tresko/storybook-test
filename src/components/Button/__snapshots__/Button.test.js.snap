// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render Button with text 1`] = `
<button
  className="sc-bdVaJa Zovcp"
  onClick={[Function]}
>
  test
</button>
`;
